main.cpp

```diff
@@ -109,20 +93,13 @@ int main()
     T t2( 7, "b");                                             //6
     
     C f;                                            //7
-    auto* smaller = f.compare(&t1, &t2);  
-    if(smaller == nullptr)
-    {
-        std::cout << "smaller cannot be equal to a nullptr" << std::endl;
-    } 
-    else
-    {
+    const auto& smaller = f.compare(t1, t2);  
         std::cout << "the smaller one is << " << smaller->name << std::endl; //9
-    }                            //8
     
```

Comment:
This code did not need revision.
`compare()` is still returning a pointer.
You are capturing that pointer as a const reference.  Temporary objects have not been covered yet. 
- [ ] revert this back to the version that checked if `smaller` is a nullptr or not.
- [ ] revise